### ISSUE TITLE
Add encryption

### ISSUE THREAD

**Issue by dkahle** [coneill-math/m2r#218]
Add encryption for data communicated through the socket to a remote host.

**Comment by dkahle** [comment]
https://stackoverflow.com/questions/2599791/best-practice-for-secure-socket-connection

**Comment by coneill-math** [comment]
I assume `R` has builtin SSH and/or RSA support?  Not sure about Macaulay2 though...

**Comment by dkahle** [comment]
I don't think so; I think we'll need to send the data through a SSH tunnel managed by the OS? Not sure. I think authentication may be a problem, especially if we want to keep this seamless.
